src/extension/infer.rs

```diff
@@ -1092,4 +1104,183 @@ mod test {
+        )?;
+
+        let [bb1, bb1_in, bb1_out] = create_with_io(
+            &mut hugr,
+            root,
+            ops::BasicBlock::DFB {
+                inputs: type_row![NAT],
+                other_outputs: type_row![],
+                predicate_variants: vec![type_row![NAT], type_row![NAT]],
+                extension_delta: b.clone(),
+            },
+        )?;
+
+        let add_bc = hugr.add_node_with_parent(
+            bb0,
+            NodeType::open_extensions(make_opaque(
+                B,
+                FunctionType::new(vec![NAT], oneway.clone()).with_extension_delta(&bc),
+            )),
+        )?;
+
+        let add_b = hugr.add_node_with_parent(
+            bb1,
+            NodeType::open_extensions(make_opaque(
+                B,
+                FunctionType::new(vec![NAT], twoway.clone()).with_extension_delta(&b),
+            )),
+        )?;
+
+        let [bb10, bb10_in, bb10_out] = create_with_io(
+            &mut hugr,
+            root,
+            ops::BasicBlock::DFB {
+                inputs: type_row![NAT],
+                other_outputs: type_row![],
+                predicate_variants: vec![type_row![NAT]],
+                extension_delta: c.clone(),
+            },
+        )?;
+
+        let [bb11, bb11_in, bb11_out] = create_with_io(
+            &mut hugr,
+            root,
+            ops::BasicBlock::DFB {
+                inputs: type_row![NAT],
+                other_outputs: type_row![],
+                predicate_variants: vec![type_row![NAT]],
+                extension_delta: c.clone(),
+            },
+        )?;
+
+        let add_c0 = hugr.add_node_with_parent(
+            bb10,
+            NodeType::open_extensions(make_opaque(
+                C,
+                FunctionType::new(vec![NAT], oneway.clone()).with_extension_delta(&c),
+            )),
+        )?;
+
+        let add_c1 = hugr.add_node_with_parent(
+            bb11,
+            NodeType::open_extensions(make_opaque(
+                C,
+                FunctionType::new(vec![NAT], oneway).with_extension_delta(&c),
+            )),
+        )?;
+
+        // Internal wiring for DFGs
+        hugr.connect(entry_in, 0, mkpred, 0)?;
+        hugr.connect(mkpred, 0, entry_out, 0)?;
+
+        hugr.connect(bb0_in, 0, add_bc, 0)?;
+        hugr.connect(add_bc, 0, bb0_out, 0)?;
+
+        hugr.connect(bb1_in, 0, add_b, 0)?;
+        hugr.connect(add_b, 0, bb1_out, 0)?;
+
+        hugr.connect(bb10_in, 0, add_c0, 0)?;
+        hugr.connect(add_c0, 0, bb10_out, 0)?;
+        hugr.connect(bb11_in, 0, add_c1, 0)?;
+        hugr.connect(add_c1, 0, bb11_out, 0)?;
+
+        // CFG Wiring
+        hugr.connect(entry, 0, bb0, 0)?;
+        hugr.connect(entry, 1, bb1, 0)?;
+        hugr.connect(bb1, 0, bb10, 0)?;
+        hugr.connect(bb1, 1, bb11, 0)?;
+
+        hugr.connect(bb0, 0, exit, 0)?;
+        hugr.connect(bb10, 0, exit, 0)?;
+        hugr.connect(bb11, 0, exit, 0)?;
+
+        hugr.infer_extensions()?;
```

Comment:
The idea is that you've fixed them, so this also checks consistency with the extensions specified already?